.github/workflows/release_mac.yml

```diff
@@ -100,12 +100,14 @@ jobs:
         twine upload --verbose dist/*.whl --repository-url https://test.pypi.org/legacy/ -u ${{ secrets.TESTPYPI_USERNAME }} -p ${{ secrets.TESTPYPI_PASSWORD }}
         # Build and upload source distribution to TestPyPI
         git clean -xdf
-        python setup.py sdist
+        python setup.py sdist --weekly_build
         twine upload dist/* --repository-url https://test.pypi.org/legacy/ -u ${{ secrets.TESTPYPI_USERNAME }} -p ${{ secrets.TESTPYPI_PASSWORD }}
         # Test the uploaded source distribution
         python -m pip uninstall -y onnx
-        python -m pip install -i https://test.pypi.org/simple/ --no-binary :all: onnx-weekly
+        python -m pip install setuptools
+        python -m pip install --index-url https://test.pypi.org/simple --use-deprecated=legacy-resolver --no-use-pep517 --no-binary onnx-weekly onnx-weekly
```

Comment:
https://github.com/pypa/pip/issues/7831 The source distribution of setuptools is missing in TestPyPI. To solve it, use old `--no-use-pep517` to not install setuptools from source and use `--use-deprecated=legacy-resolver` to resolve the package name mismatch warning (onnx-weekly vs onnx).